x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -300,6 +301,12 @@ private void createApiKeyAndIndexIt(
         Set<RoleDescriptor> userRoleDescriptors,
         ActionListener<CreateApiKeyResponse> listener
     ) {
+        if (TcpTransport.isUntrustedRemoteClusterEnabled()
```

Comment:
If the feature flag is disabled, is the code falling through this check and allowing remote cluster indices?

The PR tag `test-release` can be used to catch that type of error.

Reply:
Good q! 

This one is intentional. The purpose of this check is to allow us to defer building support for API keys to a follow up PR. I just realized that some of my explanatory comments aren't visible anymore unless you expand "hidden conversations". The relevant comment is [here](https://github.com/elastic/elasticsearch/pull/90614#discussion_r991606435).

Adding more detail:

We already enforce that `RoleDescriptor` cannot contain remote indices, inside the `parse` method, unless flag is enabled. In other words, if the feature flag is disabled remote indices will already be prevented earlier in the code.

The if statement here enforces an extra condition, namely that if the feature flag is enabled, we still prevent API keys from being created because I want to defer implementing support for those to a separate PR. If `TcpTransport.isUntrustedRemoteClusterEnabled()` is `false`, there is no need to do the extra work of iterating through role descriptors and checking if any have remote indices since we already know that can't be the case.

Reply:
Thanks for the explanation.